### PR TITLE
More streamlining, fault tolerance, and coverage increasing

### PR DESCRIPTION
**Pull request by blink1073**
87 -> 92%, and caught some bugs!

### CODE REVIEW

File: pymatbridge/pymatbridge.py [hunk 1]
```diff
@@ -169,8 +165,9 @@ def __init__(self, executable, socket_addr=None,
 
         self.context = None
         self.socket = None
+        atexit.register(self.stop)
 
-    def _program_name(self):
+    def _program_name(self):  # pramga: no cover
```
Comment (isbadawi) [inline pymatbridge/pymatbridge.py]: pragma, not pramga

File: pymatbridge/matlab_magic.py [hunk 2]
```diff
@@ -22,8 +22,6 @@
 import pymatbridge as pymat
 from .compat import text_type
 
-ipython_version = int(IPython.__version__[0])
```
Comment (arokem) [inline pymatbridge/matlab_magic.py]: Probably safer to use `IPython.version_info`

**Comment by blink1073** [comment]
Well crap, `np.asfortranarray` changes the order of the elements.

File: pymatbridge/matlab_magic.py (hunk 2 above)
Comment (blink1073) [inline pymatbridge/matlab_magic.py]: Not needed any more, I used a version-agnostic method.

**Comment by isbadawi** [comment]
There seems to be an issue with the Travis build. If the tests fail, it waits for 10 minutes and times out.

**Comment by blink1073** [comment]
@isbadawi, yeah, I'm not sure what to do about that.  The Python 3.4 version had the same behavior even when passing.

**Comment by blink1073** [comment]
We could add a kill switch alarm to make it stop sooner...

**Comment by blink1073** [comment]
Oh great, now they are all hanging...

File: pymatbridge/matlab_magic.py (hunk 2 above)
Comment (arokem) [inline pymatbridge/matlab_magic.py]: Oh yes. This is red. Silly me.

**Comment by arokem** [comment]
It looks like the test failure leaves it hanging there.

**Comment by blink1073** [comment]
Hold on before merging, I'm adding two more tests.

**Comment by blink1073** [comment]
The hanging behaviour was a result of me removing `__del__` in the magic without putting the explicit `stop()` call back in `unload_ipython_extension`.  Fixed now, and ready to merge!

**Pull request merged**